main.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  # my_each()
+  def my_each()
+    each do |item|
+      yield item if block_given?
+    end
+  end
+  # ending my_each()
+
+  # my_each_with_index()
+  def my_each_with_index()
+    index = 0
+    each do |item|
+      yield(item, index) if block_given?
+      index += 1
+    end
+  end
+  # my_each_with_index()
+
+  # my_select
+  def my_select()
+    new_arr = []
+    each do |item|
+      new_arr << item if yield(item)
+    end
+    new_arr
+  end
+  # my_select
+
+  # my_all?
+  def my_all?()
+    flag = true
+    each do |item|
+      feedback = yield(item)
+      flag = false if (feedback == false) || feedback.nil?
+    end
+    flag
+  end
+  # my_all?
+
+  # my_any
+  def my_any?()
+    flag = false
+    each do |item|
+      flag = true if yield(item)
+    end
+    flag
+  end
+  # my_any
+
+  # my_none?
+  def my_none?()
+    is_none = true
+    each do |item|
+      is_none = false if yield(item)
+    end
+    is_none
+  end
```

Comment:
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true

- [ ] my_none? when no block or argument is given returns false only if one of the collection members is true

- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class

- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric

- [ ] my_none? when a class is passed as an argument returns false if any of the collection is a member of such class

- [ ] my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex

- [ ] my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex

- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

- [ ] my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern